### ISSUE TITLE
npm install on deploy can break node_modules horribly due to bundledDependencies

### ISSUE THREAD

**Issue by yongpingchen** [jfhbrook/wzrd.in#144]
- can not get standalone npm module js file
- for example https://wzrd.in/standalone/concat-stream@latest
  - response with 
  ```
   Cannot GET /standalone/concat-stream@latest
  ```

**Comment by jfhbrook** [comment]
what the fuck? okay this is weird.

**Comment by jfhbrook** [comment]
I have no idea what happened, though it looks like the leveldown bindings got broken somehow.

I reinstalled everything and restarted. It looks like the cache is completely cold, so glhf for the next while while it desperately tries to warm up.

**Comment by jfhbrook** [comment]
It's weirder than that. I'm so confused.

Will try to update as anything becomes available.

**Comment by jfhbrook** [comment]
I think something triggered a modules rebuild and it royally screwed up. I'm doing a lot of manual adjusting to get it into a working state.

Some of these issues will go away once https://github.com/jfhbrook/wzrd.in/pull/133 hits.

**Comment by buzzdecafe** [comment]
FWIW i am getting the 502 error again.

**Comment by jfhbrook** [comment]
That's because I turned off public access while triaging the issue.

I'm able to hit the route now. Everything should be fine, For Now.

Cheers.

**Comment by buzzdecafe** [comment]
please do let me or @crosseye know if/how we can help -- we're certainly getting benefit from the service would be nice to contribute back

**Comment by jfhbrook** [comment]
Yeah? That might be a possibility. @maxogden owns the box so we'll have to work with him on how we want to manage this, but in general there's plenty of work that can be done from the ops side. For instance, it would be cool if we had some ansible stuff for box setup, systemd configs (we're using mon right now), a logrotate config, etc. Neither of us are ops people (though I want to be) and we never got to like sit down and properly line this out.

Do you do IRC? I'm in #stackvm with this handle, if you want to chat more casually about things. Max has been pretty busy the last week or but maybe he's had some bandwidth open up, dunno.

**Comment by jfhbrook** [comment]
I also have a large refactor on the table at #133 that could use some eyes, but I already got one review and, honestly, with a +/-3000 diff it's asking a lot.

**Comment by jfhbrook** [comment]
so you two work on ramda?

**Comment by buzzdecafe** [comment]
yep, ramda is our lib, and we're using wzrd to host some stuff for the web-repl. i haven't been on irc in years, but can get on again. Never used ansible, but used to use logrotate all the time, and very comfy at bash prompt

**Comment by jfhbrook** [comment]
Word.

Ansible isn't awful, especially for small scale stuff. Really all you're doing is specifying your deps with a yaml file and then handing that and an ssh conn/key to ansible and it goes to town. It can be a handy tool, especially if you want to be able to recreate the box later. I think it's also possible to test out using vagrant, which is a plus.

**Comment by jfhbrook** [comment]
but yeah, how about this: I'll try to dig into a logrotate conf this weekend, which should help with #103 , I'll work out a deploy strategy with @maxogden in the next week or so, and we can think about ansible-izing the thing later. If not you, someone linked me to a logrotate conf that I can crib from, so I think it'll be okay.

**Comment by buzzdecafe** [comment]
that was me. gl

**Comment by CrossEye** [comment]
I've also not been on IRC much in years, but can get on as necessary.  I have little ops experience, but there has often been some as part of my dev jobs, so I've picked up what I needed to.

I have no idea what ansible is, except that having read Le Guin, I can make some guesses.  And reading the link from @buzzdecafe, my guess about logrotate was correct.  (Hard to guess wrong with that name.)

It looks like everything is under control at the moment, but I really do want to be able to help if I can.  [Our REPL][rr] has been built -- as @buzzdecafe one put it -- Blanche DuBois style: "I've always relied on the kindness of strangers."  Lots of short-term drop-in contributors have helped out with it, and one of them did some really nice work that involved using wzrd, which I'd never seen before.  It's been a great tool and I'd love to give back.

  [rr]: http://ramdajs.com/repl/?v=0.22.1

**Comment by maxogden** [comment]
@CrossEye @buzzdecafe hello! If I was gonna pick the number one issue that would make wzrd.in better I think it would be https://github.com/jfhbrook/wzrd.in/issues/117. 

Another issue is how we have to globally upgrade browserify, which breaks backwards compatibility. Here's an issue I just made for that: https://github.com/jfhbrook/wzrd.in/issues/146

Additionally theres just general maintenance on the wzrd.in server code itself, if anyone has express chops it would be great if you wanted to dig in and start fixing some small bugs
